VideoLocker/src/main/java/org/edx/mobile/http/HttpRequestDelegate.java

```
@@ -48,19 +50,24 @@ public T fromJson(String json) throws Exception{
 
     public T fetchData(boolean preferCache) throws Exception{
         String json = null;
-        String cacheKey = getCacheKey();
+        String cacheKey = endPoint.getCacheKey();
         if ( preferCache  || !NetworkUtil.isConnected(MainApplication.instance()) ){
             try {
                 json = cacheManager.get(cacheKey);
             } catch (Exception e) {
                 logger.error(e);
             }
-            if ( json != null )
+            if ( json != null ) {
                 try {
                     return fromJson(json);
                 } catch (Exception e) {
                     logger.error(e);
                 }
+            }
+        }
+        //if it is on the UI thread, we just can not make http call
+        if (Looper.getMainLooper().getThread() == Thread.currentThread()){
```

This seems like a really surprising thing to silently fail?